core/src/main/java/org/infinispan/globalstate/impl/GlobalConfigurationManagerImpl.java

```diff
@@ -0,0 +1,195 @@
+package org.infinispan.globalstate.impl;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.lang.invoke.MethodHandles;
+import java.nio.channels.FileLock;
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.infinispan.Cache;
+import org.infinispan.commands.RemoveCacheCommand;
+import org.infinispan.commons.CacheConfigurationException;
+import org.infinispan.commons.api.CacheContainerAdmin;
+import org.infinispan.commons.util.concurrent.ConcurrentHashSet;
+import org.infinispan.configuration.cache.Configuration;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.configuration.global.GlobalConfiguration;
+import org.infinispan.configuration.parsing.ConfigurationBuilderHolder;
+import org.infinispan.configuration.parsing.ParserRegistry;
+import org.infinispan.factories.annotations.Inject;
+import org.infinispan.factories.annotations.PostStart;
+import org.infinispan.factories.annotations.Start;
+import org.infinispan.globalstate.GlobalConfigurationManager;
+import org.infinispan.globalstate.ScopeFilter;
+import org.infinispan.globalstate.ScopedState;
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.registry.InternalCacheRegistry;
+import org.infinispan.util.TimeService;
+import org.infinispan.util.logging.Log;
+import org.infinispan.util.logging.LogFactory;
+
+/**
+ * Implementation of {@link GlobalConfigurationManager}
+ *
+ * @author Tristan Tarrant
+ * @since 9.2
+ */
+
+public class GlobalConfigurationManagerImpl implements GlobalConfigurationManager {
+   static Log log = LogFactory.getLog(MethodHandles.lookup().lookupClass());
+   private GlobalConfiguration globalConfiguration;
+   private EmbeddedCacheManager cacheManager;
+   private Cache<ScopedState, Object> stateCache;
+   private GlobalConfigurationStateListener stateCacheListener;
```

Comment:
I guess it can be removed... it is local only... or the `stop()` method is missing to remove the listener :)